xen-bugtool

```diff
@@ -557,8 +573,68 @@ def output(x):
 def output_ts(x):
     output("[%s]  %s" % (time.strftime("%x %X %Z"), x))
 
+def is_deemed_executable(command):
+    # type: (str | list[str]) -> bool
+    """Check if command is executable (supports absolute paths and it looks in $PATH)
+    :param command: The command to check, either a string or a list of strings.
+
+    Note: This function is also called from load_plugins(), feeding it a string
+    that contains arbitrary commands, shell builtins and even loops. An example
+    is `ls -lR /lib/modules/$(uname -r)/updates`. Such strings are not
+    validated here, and are only checked for being non-empty.
+    """
+
+    def path_is_absolute(cmd_path):  # type: (str) -> bool
+        """Check if cmd_path is an absolute path"""
+        return cmd_path.startswith("/")
+
+    def path_is_executable(cmd_path):  # type: (str) -> bool
+        """Check if cmd_path is an executable file"""
+        # isfile() follows symlinks, so this works for symlinked commands as well
+        return os.path.isfile(cmd_path) and os.access(cmd_path, os.X_OK)
+
+    if not isinstance(command, list):
+        # We have a string, command is the string command or script to check
+        if path_is_absolute(command):
+            # plugin script starts with an absolute path, return its executable state
+            return path_is_executable(command.split()[0])
+
+        # We cannot validate other strings, so check those just for non-emptiness:
+        # The load_plugins() function calls this function with arbitrary strings
+        # that may contain shell builtins, loops, etc. We cannot validate those,
+        # so we just check that the string is not empty or only whitespace.
+        return bool(command and command.strip())
+
+    # We have a list of command arguments, command[0] is the command to check
+    if path_is_absolute(command[0]):
+        return path_is_executable(command[0])
+
+    # command[0] was not an absolute path, look for the command in $PATH
+    for path_part in os.environ.get("PATH", "").split(":"):
+        # For xen-bugtool we only accept absolute paths here as we don't want
+        # to execute anything from the current directory (".") or relative paths
+        # in $PATH: This would pose a risk to the predictability of the command.
+        if path_part.startswith("/"):
```

Comment:
can you not just call `path_is_absolute(path_part)` here?

Reply:
@MarkSymsCtx thanks, done!